### PR TITLE
Linter: Check condition IDs are unique

### LINKED ISSUE #70: no lint error for non-unique condition ids
`petab.lint.check_condition_df` does not check whether index entries are unique, but this is required according to documentation.

### PR DESCRIPTION
**Pull request by dweindl**
Closes #70

### CODE REVIEW

File: petab/lint.py [hunk 1]
```diff
@@ -108,6 +108,10 @@ def check_condition_df(
 
     check_ids(df.index.values, kind='condition')
 
+    if not df.index.is_unique:
```
Comment (dilpath) [inline petab/lint.py:111]: Maybe this is inappropriate according to the docs, and could be replaced with `if df.index.has_duplicates`
```python
>>> print(pd.Index.is_unique.__doc__)

        Return if the index has unique values.
        
>>> print(pd.Index.has_duplicates.__doc__)

        Check if the Index has duplicate values.
```

Comment (dilpath) [reply petab/lint.py:111]: The tests may have passed because there are always only duplicated or only unique values in any tested index (no mixture).

**Review by dilpath — APPROVED** [review]
:+1:

File: petab/lint.py (hunk 1 above)
Comment (dweindl) [reply petab/lint.py:111]: I think the docstring there maybe slightly misleading. `has_duplicates` is only `return not self.is_unique `. So, I'd directly use the latter.

**Pull request merged**